doc/src/location_tracking.md

```diff
@@ -0,0 +1,25 @@
+# Location Tracking
+
+Sometimes it may be helpful to know where in the input stream a particular
+token was encountered. For example, this can be useful when generating
+user-visible messages that reference specific points in the input.
+
+This is achieved via the location tracking macros, `@L` and `@R`.  `@L` looks
+ahead to the token immediately to its right, and binds the the location (in
+bytes) in the input stream where that token starts.  Similarly, `@R` binds the
+location where the token immediately to the left ends.
```

Comment:
```suggestion
This is achieved via the location tracking macros, `@L` and `@R`.  `@L`
binds the start position - in bytes - of the token immediately to its right
in the input stream. Similarly, `@R` binds the index of the last byte _plus
one_ of the token immediately to its left.
```

I tried to reflow this paragraph a bit, as I find it slightly convoluted, but it's just a suggestion.

Reply:
Thanks for the reword, and the catch about "plus one".  I added a paragraph and example at the end to show how that integrates cleanly with Range, since I wonder if people will worry that the plus one is something they need to directly account for.  Let me know if that all makes sense to you.